iroh-sync/src/store/fs/bounds.rs

```diff
@@ -0,0 +1,169 @@
+use std::ops::Bound;
+
```

Comment:
all these bounds are quite important, I wonder if we could write some proptests to make sure they behave as expected

Reply:
One a first thought I can't find a way to write meaningful proptests here. Do you have an idea how they should look or what they should test for?

Reply:
I don't 😓 mostly thinking out loud

Reply:
I added some manual tests for the edgecases.